### ISSUE TITLE
Links: drop complex functionality - splitter maps

### ISSUE THREAD

**Issue by breznak** [htm-community/nupic.cpp#257]
https://github.com/htm-community/nupic.cpp/issues/253#issuecomment-462052482 

>remove.. Link complexity then we just remove the Splitter maps and the LinkPolicy stuff that creates them. Otherwise, the links are not complicated.

I think this would be reasonable API breaking change.

**Comment by dkeeney** [comment]
_**API Docs**_
I went back to the API documentation to see what it actually says about splitter maps.
The only reference is this one in a side note:
```
A Link contains a LinkPolicy, also not described here, which is responsible for generating the splitter map.
```
In fact, I think this is the only  reference to `LinkPolicy`  and nowhere does it define what it is or how to use it.

_**So, would I break any existing code?**_
Only the Unit tests it seems.  In some of the examples, the Link( ) call passes "UniformLink" as the link policy name.  This is the default link policy which effectively does nothing but pass data unchanged.

I was not able to find any other use of LinkPolicy anywhere in the .py code or C++ code other than in the unit tests.  So this is all Code Rot.

So, I propose we remove it. This involves the following files:
- LinkPolicy.hpp     REMOVE
- LinkPolicyFactory.cpp   REMOVE
- LinkPolicyFactory.hpp  REMOVE
- TestFanIn2LinkPolicy.cpp   REMOVE
- TestFanIn2LinkPolicy.hpp  REMOVE
- UniformLinkPolicy.cpp REMOVE
- UniformLinkPolicy.hpp REMOVE
- UniformLinkPolicyTest.cpp  REMOVE
- Link.cpp   MODIFY Remove Linktype and linkParams implementation but accept "UniformLink" as the linktype argument and then ignore it.
- Input.cpp MODIFY Remove getSplitterMap() and getInputForNode( ) which use splitter maps.
- InputTest.cpp  MODIFY to remove InputTest.SplitterMap and modify InputTest.LinkTwoRegionsOneInput to work without splitter map.
- TestNode.cpp  MODIFY compute( ) to remove call to getInputForNode( ) which uses splitter map.  This is the only region implementation I can find that uses it.

**Comment by dkeeney** [comment]
Stepping back and looking at this from a bigger picture...

the splitter maps (and the LinkPolicy that creates them) were used as a way to re-arrange bits in the input buffer based on user defined patterns.  However, thinking about how this works in biology, the neurons make connections to the synapse of other neurons. There is no order or pattern involved and in fact it needs to be fairly random as to how the neurons connect.  It is the job of the Spacial Pooler to simulate that set of random connections.  So in effect, any re-mapping of bits by a splitter map prior to being presented to the Spacial Pooler should have no affect on functionality.

So I suspect this is why we do not see any use of this capability anywhere in the .py code.

**Comment by ctrl-z-9000-times** [comment]
See PR #271

**Comment by breznak** [comment]
Closing as fixed in #271
